ipn/ipnlocal/local.go

```diff
@@ -3872,7 +3884,18 @@ func (b *LocalBackend) resetControlClientLockedAsync() {
 		b.numClientStatusCalls.Add(1)
 	}
 
-	go b.cc.Shutdown()
+	if testenv.InTest() {
+		// From 2021-04-20 to 2023-08-31 we always did this shutdown
+		// asynchronously. But tests flaked on it sometimes, as our
+		// tests often do resource leak checks at the end to make sure
+		// everything's shut down. So do this synchronously in tests
+		// to deflake tests.
+		//
+		// TODO(bradfitz,maisem): do this always?
+		b.cc.Shutdown()
```

Comment:
i think we should just do synchronously always and deal with the fallout.

Reply:
I think previously we were doing this async in fears that it might block, but i don't think it can.